src/NServiceBus.Core/Causation/AttachCausationHeadersBehavior.cs

```diff
@@ -28,7 +28,7 @@ static void SetRelatedToHeader(IOutgoingPhysicalMessageContext context, Incoming
             context.Headers[Headers.RelatedTo] = incomingMessage.MessageId;
         }
 
-        static void SetConversationIdHeader(IOutgoingPhysicalMessageContext context, IncomingMessage incomingMessage)
+        static void SetConversationIdHeader(IOutgoingLogicalMessageContext context, IncomingMessage incomingMessage)
```

Comment:
While you're here, how about switching those out parameters below to out var instead?